Assets/HoloToolkit/Utilities/Scripts/CanvasRaycastResult.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace HoloToolkit.Unity
+{
+    public struct CanvasRaycastResult : IComparable<CanvasRaycastResult>
+    {
+        public readonly RaycastResult RaycastResult;
+        public readonly Canvas Canvas;
+
+        public CanvasRaycastResult(RaycastResult raycastResult)
+        {
+            RaycastResult = raycastResult;
+            Canvas = raycastResult.gameObject != null ? raycastResult.gameObject.GetComponentInParent<Canvas>() : null;
```

Comment:
Can we confirm that this `GetComponent` isn't being called each update?

Reply:
It is being called each update for all raycast target UI elements in its path. This would be 1-3 elements in a properly set up UI.

Reply:
It seems like we're just getting the component to check to see if it's the same depth.  

Is there another way we can do this?  This will impact performance too much.